AutomatedTesting/cmake/EngineFinder.cmake

```diff
@@ -7,35 +7,72 @@
 #
 #
 # {END_LICENSE}
-# This file is copied during engine registration. Edits to this file will be lost next
-# time a registration happens.
+# Edits to this file may be lost in upgrades. Instead of changing this file, use 
+# the 'engine_finder_cmake_path' key in your project.json or user/project.json to specify 
+# an alternate .cmake file to use instead of this one.
 
 include_guard()
 
-# Read the engine name from the project_json file
+include(cmake/VersionUtils.cmake)
+
+set(ENGINE_FINDER_VERBOSE FALSE CACHE BOOL "Increase EngineFinder.cmake verbosity.  Useful for debugging engine selection based on compatibility.")
+
```

Comment:
This isn't needed.
CMake already has [--log-level](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-log-level) and what we should do is make the messages `VERBOSE` using that keywork.

With using a Cache variable, that value is now sticky and setting the verbosity will stick between subsequent runs until the user changes the value back to FALSE